google/cloud/spanner_v1/pool.py

```diff
@@ -199,21 +203,47 @@ def bind(self, database):
                 _metadata_with_leader_aware_routing(database._route_to_leader_enabled)
             )
         self._database_role = self._database_role or self._database.database_role
-        request = BatchCreateSessionsRequest(
-            database=database.name,
-            session_count=self.size - self._sessions.qsize(),
-            session_template=Session(creator_role=self.database_role),
-        )
 
-        while not self._sessions.full():
-            resp = api.batch_create_sessions(
-                request=request,
-                metadata=metadata,
+        with trace_call("CloudSpanner.BatchCreateSessions", self) as span:
+            requested_session_count = self.size - self._sessions.qsize()
+            request = BatchCreateSessionsRequest(
+                database=database.name,
+                session_count=requested_session_count,
+                session_template=Session(creator_role=self.database_role),
+            )
+
+            if requested_session_count > 0:
+                current_span.add_event(
```

Comment:
current_span is not initialized.

Reply:
Sheesh, sorry I had a bunch of benchmarks running locally and they interrupted my `nox -s unit-3.8` tests so didn't see those but all fixed now.